### DISCUSSION TITLE
react-native-image-marker:compileDebugKotlin FAILED

### DISCUSSION THREAD

**Discussion by Tiepbm** [JimmyDaddy/react-native-image-marker#175] [Q&A] [answered]
**Describe the bug**
e: file:///Users/tiepbm/Desktop/TiepBM/PVI/PVI247/node_modules/react-native-image-marker/android/src/main/java/com/jimmydaddy/imagemarker/ImageLoader.kt:140:34 Type mismatch: inferred type is ImagePipelineExperiments but ImagePipelineConfig? was expected
![Uploading image.png…]()

**Devlopment environment(please complete the following information):**
 - OS: Mac M1
 - nodejs:  21.2.0
 - react-native: 0.73.0
 - react-native-image-marker :  1.1.10

**Smartphone (please complete the following information):**
 - Device:SamSung S22
 - OS: Android 12

thanks!

**Comment by JimmyDaddy** [comment]
@Tiepbm the image didn't upload successfully. By the way what is the `minSdkVersion` of your main android project(you can check it in `${your app root dir}/android/build.gradle`)

**Comment by Tiepbm** [comment]
> @Tiepbm the image didn't upload successfully. By the way what is the `minSdkVersion` of your main android project(you can check it in `${your app root dir}/android/build.gradle`)
 buildToolsVersion = "34.0.0"
        minSdkVersion = 21
        compileSdkVersion = 34
        targetSdkVersion = 34
        ndkVersion = "25.1.8937393"
        kotlinVersion = "1.6.20" //old version is 1.8
![e5357eb85bccf392aadd](https://github.com/JimmyDaddy/react-native-image-marker/assets/3059239/5f25f7ba-0557-4ab7-be59-3aa6699851c2)

**Comment by JimmyDaddy** [comment]
@Tiepbm I have found the cause of the problem. You should upgrade your kotlinVersion to `1.7.0` or newer (`1.8.0` is okay too).

**Reply by Tiepbm** [reply]
i upgraded my kotlinVersion 1.7.0 or 1.8.0. But error still. :(
this is my android/build.gradle
<img width="638" alt="image" src="https://github.com/JimmyDaddy/react-native-image-marker/assets/3059239/fd6dbffb-ae30-4aa2-b047-d4117853eff0">

**Reply by JimmyDaddy** [reply]
@Tiepbm 
Can you tell me what value the `FLIPPER_VERSION` is in your project? You can check it in `${your project root dir}/android/gradle.properties`

You can try the version `FLIPPER_VERSION=0.125.0`

**Reply by Tiepbm** [reply]
@JimmyDaddy i update react native from 0.70.6 to 0.73.0
They said that delete FLIPPER_VERSION in android/gradle.properties:
![image](https://github.com/JimmyDaddy/react-native-image-marker/assets/3059239/ca2b1ab6-a68b-41c7-b5c1-cbb502d95966)
and android/app/build.gradle: 
![image](https://github.com/JimmyDaddy/react-native-image-marker/assets/3059239/dcdbb7b0-f7cd-4782-bee5-b4d73d14fa78)

**Reply by JimmyDaddy** [reply]
Did you use `com.facebook.flipper:flipper-fresco-plugin` in any other plugin?

**Reply by EducoreAPK** [reply]
Having same issue

**Reply by Tiepbm** [reply]
no, i used implementation("com.facebook.react:flipper-integration")

**Reply by JimmyDaddy** [reply]
I have found the real cause of this problem, [RN <= 0.73.0 uses fresco version is `2.xx.x`  ](https://github.com/facebook/react-native/blob/8cf9505bd27c5dade33b17cc177fa5ef1613dbcd/ReactAndroid/gradle.properties#L15)
[But 0.73.0 uses `3.1.3`](https://github.com/facebook/react-native/blob/43826facfab8eed7d01a801778d1c477e60730a6/packages/react-native/gradle/libs.versions.toml#L19C1-L19C1).
It seems `ImagePipelineConfig.newBuilder(context).experiment().setMaxBitmapSize(int).build()` return `ImagePipelineExperiments` instead of `ImagePipelineConfig` now. 
Sorry for that, I'll update a fix ASAP to adapt the `0.73.x`

**Comment by JimmyDaddy** [comment, marked as answer]
@Tiepbm @EducoreAPK, v1.2.0 is now available. I've replaced Fresco with [Coil](https://github.com/coil-kt/coil) for image loading on Android #177 . If you want to use v1.2.0, you'll need to upgrade your React Native version to 0.73.0 or newer. It's not compatible with versions 0.72.x or older.

**Reply by MoinVnr** [reply]
V1.2.0 and v1.2.6 both i tried but not working for Adding watermark image to camera capture using react-native-camera
React Native version to 0.73.6
 <RNCamera
        style={{ flex: 1 }}
        type={RNCamera.Constants.Type.back}
        ref={(ref) => setCameraRef(ref)}>
        
        {markerVisible && (
          <ImageMarker
            source={{ uri: latestImageURI }}
            style={{
              position: 'absolute',
              top: '2%',
              left: '80%',
              width: 50,
              height: 50,
              resizeMode: 'contain'
            }}
            markerSource={markerLogo}
            // onEnd={(uri) => {
                    setMarkerVisible(false);
             }}
          />
        )}
  </RNCamera>
  
  Error :  Error taking picture: [Error: Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: undefined. You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.

Check the render method of `CameraScreen`.]
